cpp/src/merge/merge.cu

```diff
@@ -0,0 +1,489 @@
+                        true,
+                        stream_);
+
+    //set the null count:
+    //
+    p_merged_col->set_null_count(lcol.null_count() + rcol.null_count());
+
+    //to resolve view.data()'s types use: ElemenT
+    //
+    ElemenT const* p_d_lcol = lcol.data<ElemenT>();
+    ElemenT const* p_d_rcol = rcol.data<ElemenT>();
+        
+    auto exe_pol = rmm::exec_policy(stream_);
+    
+    //capture lcol, rcol
+    //and "gather" into merged_view.data()[indx_merged]
+    //from lcol or rcol, depending on side;
+    //
+    thrust::transform(exe_pol->on(stream_),
+                      dv_row_order_.begin(), dv_row_order_.end(),
+                      merged_view.begin<ElemenT>(),
+                      [p_d_lcol, p_d_rcol] __device__ (index_type const& index_pair){
+                       auto side = thrust::get<0>(index_pair);
+                       auto index = thrust::get<1>(index_pair);
+                       
+                       ElemenT val = (side == side::LEFT ? p_d_lcol[index] : p_d_rcol[index]);
+                       return val;
+                      }
+                     );
+
+    //cudaDeviceSynchronize();//? nope...these two could proceed concurrently
+
+
+    //CAVEAT: conditional call below is erroneous without
+    //set_null_mask() call (see TODO above):
+    //
+    if (lcol.has_nulls() || rcol.has_nulls())
+      //resolve null mask:
+      //
+      materialize_bitmask(lcol,
+                          rcol,
+                          merged_view,
+                          dv_row_order_.data().get(),
+                          stream_);
+                   
+    return p_merged_col;
+  }
+
+  //specialization for string...?
+  //or should use `cudf::string_view` instead?
+  //
+  template<typename ElemenT>//required: column type
+  std::enable_if_t<not cudf::is_fixed_width<ElemenT>(),
+                   std::unique_ptr<cudf::column>>
+  operator()(cudf::column_view const& lcol, cudf::column_view const& rcol) const
+  {
+    //for now...
+    CUDF_FAIL("Non fixed-width types are not supported");
+
+    // TODO: separate PR for strins support:
+    //
+    // auto column = strings::detail::merge( strings_column_view(lcol),
+    //                                       strings_column_view(rcol),
+    //                                       dv_row_order_.begin(),
+    //                                       dv_row_order_.end(),
+    //                                       mr_,
+    //                                       stream_);
+    
+    // if (lcol.has_nulls() || rcol.has_nulls())
+    //   {
+    //     auto merged_view = column->mutable_view();
+    //     materialize_bitmask(lcol,
+    //                         rcol,
+    //                         merged_view,
+    //                         dv_row_order_.data().get(),
+    //                         stream_);
+    //   }
+    // return column;
+  }
+
+private:
+  VectorI const& dv_row_order_;
+  rmm::mr::device_memory_resource* mr_;
```

Comment:
I don't think this mr is ever used inside this struct. And I think no need to store it, just make it a parameter to `operator()`.

```suggestion
```

Reply:
Yes, it will be used in the strings version operator. In addition to that, `operator()` has the semantics of a binary operator (merging 2 columns), so adding additional parameters to it would alter that semantics.

Reply:
We can type dispatch varargs.